### PR TITLE
[DNM] Log deltas for create_message

### PR DESCRIPTION
**Pull request by Mothblocks**
## About The Pull Request
Logs deltas for `/datum/tgui_window/proc/send_message` into tgui_delta.log.

This is on DNM as I don't want to add any extra processing to send_message, at least until we bring it down from consistently one of the largest overtime procs.

Queues logs until SStgui fire, and only after all tgui inputs are sent out. This necessarily comes with MC_TICK_CHECK as to make sure it's only done when we have time.

### CODE REVIEW

File: code/modules/tgui/tgui_window.dm [hunk 1]
```diff
@@ -209,7 +209,13 @@
 /datum/tgui_window/proc/send_message(type, payload, force)
 	if(!client)
 		return
+
+	var/tick_usage = TICK_USAGE
 	var/message = TGUI_CREATE_MESSAGE(type, payload)
+	var/delta = TICK_USAGE - tick_usage
+
+	SStgui.queued_calculate_deltas += list(list(message, TICK_USAGE_TO_MS(delta)))
```
Comment (MrStonedOne) [inline code/modules/tgui/tgui_window.dm:217]:
```suggestion
	SStgui.queued_calculate_deltas += list(list(message, TICK_DELTA_TO_MS(delta)))
```
TICK_USAGE_TO_MS is for when you want to pass the old tick_usage and use the current tick_usage to generate a delta

Comment (Mothblocks) [reply code/modules/tgui/tgui_window.dm:217]: whoops, i had this open too

**Comment by Mothblocks** [comment]
The extra processing power is just adding to a list...it might be fine aside from RAM, in which case we can have an upper bound where once met we just perform the logging anyway (perhaps not 100% of the list).

**Comment by Mothblocks** [comment]
Okay, probably don't merge this not because of processing power, but because it creates HUUUGE log files.

**Comment by Mothblocks** [comment]
Don't need any more code changes on this, and we can TM closed PRs.

**Pull request closed without merging**